### ISSUE TITLE
Still your preferred solution for CI and Drupal 8 w/ Behat? 

### ISSUE THREAD

**Issue by justinlevi** [geerlingguy/drupalci-sonar-jenkins#14]
@geerlingguy Jeff, is this still your preferred solution for CI, testing, and Acquia integrated with your drupal-vm (https://github.com/geerlingguy/drupal-vm) for development? 

Would you run your behat tests on the Jenkins server as well? What about Visual Regression tests? Also, how do you handle build and regression test fails to the dev team, to the client, etc? 

Email? Notification Message at the top of the dev drupal site? 

Thanks for any direction on this stuff.

**Comment by geerlingguy** [comment]
Hehe, quite the set of questions!

Here's the short of it: every team I've worked on, and every project I've worked on, has had wildly different requirements, wildly different environments, and wildly different chosen 'best practices'.

For most of my smaller Drupal sites, I _don't_ have any automated tests. No visual regressions, no Jenkins, no deploy pipeline, no special frontend tooling, nothing. I always use Git for the codebase, but that's about the only consistent thing on every project I work on.

I still use Jenkins for _most_ projects that do need CI. However, if a project can afford to pay for a Travis CI account, that's been easier to get going and integrate with the project itself (sometimes... not always—Jenkins is more flexible for sure).

For testing, I haven't worked on a project with visual regression tests in place for a couple years now, at least not with Behat/Selenium/etc.—most of the projects that need that level of automated testing either outsource it or use a for-pay tool.

Therefore I usually leave it up to individuals/teams to find what's best for them. Often it's Behat + Selenium. Sometimes it's something else. Sometimes it's not tests at all (think short timeline and brochureware site... who needs to add in the overhead of non-manual testing?).

I started this project a few years back, and it's still a great solution, but not necessarily for every project. And it leaves out a lot in terms of what some people would call the "hipster modern frontend regression testing toolkit", but I'm okay with that... it's not my wheelhouse, and I just build the base components and let others work on those parts.
